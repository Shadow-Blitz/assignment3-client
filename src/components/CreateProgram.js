import React, { Component } from 'react'
import { api, setJwt } from '../api/init'
import { Paper, Button, TextField } from '@material-ui/core'

class CreateProgram extends Component {


    handleSubmit = (e) => {
        const token = localStorage.getItem("token")
        setJwt(token)
        e.preventDefault()

        try {
            const { name, description, len, date } = e.target.elements
            console.log(len.value)
            const length = parseInt(len.value)
      if (isNaN(length) || length < 1 || length > 120) {
        alert('Length must be between 1 and 120 minutes and digits only')
        return
      }
            api.post('/programs', {
                name: name.value,
                description: description.value,
                date: date.value,
                length: len.value
            })
                .then((response) => {
                    this.props.history.push(`/programs/${response.data._id}/updateactivities`)
                })

        }
        catch (error) { console.error(error) }
    }

    render() {

        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    {console.log(this.props.history)}
                    <h1>Create a new program for your unit</h1>
                    <TextField
                        required
                        id="name"
                        label="Name"
                        margin="normal"
                        type="name"
                    />
                    <br />
                    <TextField
                        required
                        id="description"
                        label="Description"
                        margin="normal"
                        type="description"
                    />
                    <br />
                    <p>Date</p>
                    <TextField
                        required
                        id="date"
                        // label="Date"
                        margin="normal"
                        type="date"
                    />
                    <br />
                    <TextField
                        required
                        id="len"
                        label="Length"
                        margin="normal"
                        type="length"
                    />
                    <p>Attachments</p>
                    <Button type="submit" variant='contained' color="primary" style={{ 'backgroundColor': 'orange' }}>Create</Button>
                </form>
            </div >

        )
    }
}

export default CreateProgram