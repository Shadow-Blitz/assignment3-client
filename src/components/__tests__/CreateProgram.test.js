import React from 'react';
import CreateProgram from '../CreateProgram';
import { shallow } from 'enzyme';

describe("CreateProgram", () => {
    it("should render correctly", () => {
        const component = shallow(<CreateProgram />);
    });
    it("should render initial layout", () => {
        // when
        const component = shallow(<CreateProgram />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
});