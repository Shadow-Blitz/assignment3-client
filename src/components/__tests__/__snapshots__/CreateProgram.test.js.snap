// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProgram should render initial layout 1`] = `
Array [
  <div>
    <form
      onSubmit={[Function]}
    >
      <h1>
        Create a new program for your unit
      </h1>
      <TextField
        id="name"
        label="Name"
        margin="normal"
        required={true}
        select={false}
        type="name"
        variant="standard"
      />
      <br />
      <TextField
        id="description"
        label="Description"
        margin="normal"
        required={true}
        select={false}
        type="description"
        variant="standard"
      />
      <br />
      <p>
        Date
      </p>
      <TextField
        id="date"
        margin="normal"
        required={true}
        select={false}
        type="date"
        variant="standard"
      />
      <br />
      <TextField
        id="len"
        label="Length"
        margin="normal"
        required={true}
        select={false}
        type="length"
        variant="standard"
      />
      <p>
        Attachments
      </p>
      <WithStyles(Button)
        color="primary"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="submit"
        variant="contained"
      >
        Create
      </WithStyles(Button)>
    </form>
  </div>,
]
`;
